llvm/include/llvm/Target/GlobalISel/Combine.td

```diff
@@ -1094,7 +1112,9 @@ def funnel_shift_combines : GICombineGroup<[funnel_shift_from_or_shift,
                                             funnel_shift_to_rotate,
                                             funnel_shift_right_zero,
                                             funnel_shift_left_zero,
-                                            funnel_shift_overshift]>;
+                                            funnel_shift_overshift,
```

Comment:
In addition to adding to `funnel_shift_combines` as you did here, each target also selects a set of combine rules they want to use. For instance, RISCV does this in `lib/Target/RISCV/RISCVCombine.td` and `funnel_shift_combines` is not one of them. So before writing a test, you need to import `funnel_shift_combines`.

Reply:
> I assume you meant that I should include the funnel shift combines during the legalization phase? Should I include the funnel shift combines in RISCVPostLegalizerCombiner

Oh, I forgot that RISC-V runs all the combiner rules pre-legalizations. And I think your combiner rule is only useful pre-legalized since RISC-V doesn't really have native support for funnel shifts, so they'll be expanded by the legalizer: https://github.com/axelcool1234/llvm-project/blob/f1f8463c457e4b2e08a53aea62b9a47490815882/llvm/lib/Target/RISCV/GISel/RISCVLegalizerInfo.cpp#L193

Bottom line: I think you're good, no need to explicitly add `funnel_shift_combines` into RISCVCombine.td